### DISCUSSION TITLE
Remote server and local run (or vice versa)

### DISCUSSION THREAD

**Discussion by dmsfabiano** [dagster-io/dagster#15854] [Q&A]
I was wondering if there was a way to have, either:

- Dagster server is running on a remote server, and the tasks are queued there but executed on a local machine; OR
- Dagster is in a remote server but has code locations to a local machine

What we have right now:
- We have CI/CD pipelines that execute a dagster run for model training every time there is a git push

The above approach is nice, but it has the overhead of having to push, and build a docker image for every single model train we want to experiment with (even if its one parameter change). Therefore, I was wondering if there was a way to separate the user code to a local location from the remote server for quick experimentation that is still centrally supervised by the orchestrator

**Comment by smackesey** [comment]
cc @gibsondan

**Reply by prha** [reply]
@dmsfabiano Could you share images by having the parameter changes be made via run config for a shared job?

**Reply by dmsfabiano** [reply]
When multiple jobs will be ran in the cluster, yes, but what we were looking to do here is to have a central orchestrator running in our cluster and offload the run to be computed in our local computer for local runs, and report back to the orchestrator. This is the setup we had with Luigi, and it worked well for doing ML experiments without having to push to trigger the CI/CD (this way only the selected experiment is pushed on the feature branch ~ or at least less pushes happen)

I have seen it nowadays with the concept of [local agents](https://clear.ml/docs/latest/docs/clearml_agent/)
